src/irmin/node.ml

```diff
@@ -103,40 +125,66 @@ struct
   let remove t k = StepMap.remove k t
   let default = M.default
 
-  let value_t =
-    let open Type in
-    variant "value" (fun n c x -> function
-      | `Node h -> n h
-      | `Contents (h, m) -> if equal_metadata m M.default then c h else x (h, m))
-    |~ case1 "node" H.t (fun k -> `Node k)
-    |~ case1 "contents" H.t (fun h -> `Contents (h, M.default))
-    |~ case1 "contents-x" (pair H.t M.t) (fun (h, m) -> `Contents (h, m))
-    |> sealv
+  module Pre_hash = struct
```

Comment:
this submodule is necessary only because of https://github.com/mirage/repr/issues/39